openslides_backend/migrations/migration_handler.py

```diff
@@ -1,160 +1,355 @@
+            ), "Relationkind must be a normal view to create a view copy."
+            viewdef = table_re.sub(replace_suffix, viewdef)
+            self.cursor.execute(
+                sql.SQL("CREATE VIEW {view_m} AS {viewdef};").format(
+                    view_m=sql.Identifier(migration_view),
+                    viewdef=sql.Identifier(viewdef),
+                )
+            )
+            self.cursor.execute(
+                sql.SQL("INSERT INTO {table_m} SELECT * FROM {table_t};").format(
+                    table_m=table_m, table_t=table_t
+                )
+            )
+            # TODO rereference all models pointing to this collection
+        # TODO CodeBlocks probably needs a reset of generated trigger names
+        (
+            pre_code,
+            table_name_code,
+            view_name_code,  # Should be used for reads of migration. So original needs to be subbstituted with shadow table names
+            alter_table_code,
+            final_info_code,
+            missing_handled_attributes,
+            im_table_code,  # TODO check the rereference of this also
+            create_trigger_partitioned_sequences_code,  # TODO ALTER SEQUENCE to not be deleted when owner column gets deleted
+            create_trigger_1_1_relation_not_null_code,
+            create_trigger_relationlistnotnull_code,
+            create_trigger_unique_ids_pair_code,
+            create_trigger_notify_code,
+            errors,
+        ) = GenerateCodeBlocks.generate_the_code()
+        sql_text = (
+            # + alter_table_code TODO are these foreign key constraints copied over or not?
+            create_trigger_1_1_relation_not_null_code
+            + create_trigger_relationlistnotnull_code
+            + create_trigger_unique_ids_pair_code
+        )
+        replaced_blocks = []
+        trigger_re = re.compile(
+            r"(CREATE\s+(?:CONSTRAINT\s+)?TRIGGER\b.*?;)", re.IGNORECASE | re.DOTALL
+        )
 
-class MigrationCommand(StrEnum):
-    MIGRATE = "migrate"
-    FINALIZE = "finalize"
-    RESET = "reset"
-    CLEAR_COLLECTIONFIELD_TABLES = "clear-collectionfield-tables"
-    STATS = "stats"
-    PROGRESS = "progress"
+        for match in trigger_re.finditer(sql_text):
+            block = match.group(0)
+            if table_re.search(block):
+                modified_block = table_re.sub(replace_suffix, block)
+                replaced_blocks.append(modified_block)
+        sql_text = "".join(replaced_blocks)
+        self.cursor.execute(sql_text)
 
+    def execute_migrations(self) -> None:
+        """
+        Executes the data_definition and data_manipulation methods of the migrations
+        stored in MigrationHelper.migrations.
 
-class MigrationHandler(BaseHandler):
-    lock = Lock()
-    migration_running = False
-    migrate_thread_stream: StringIO | None = None
-    migrate_thread_stream_can_be_closed: bool = False
-    migrate_thread_exception: Exception | None = None
+        Returns:
+        - None
```

Comment:
Is this part of the description necessary if you already have the type annotation on the function?